server/routes/wallet.js

```diff
@@ -248,10 +248,24 @@ router.post('/public-api/buy-address', handler(async (req, res) => {
   } = req.body
   const processor = await plugins.payment
 
-  let ipAddress = req.headers[process.env.IP_HEADER_PROP_NAME] || ''
+  let ipAddress = ''
+  if (Boolean(process.env.TRUST_PROXY) && process.env.IP_HEADER_PROP_NAME) {
+    ipAddress = req.headers[process.env.IP_HEADER_PROP_NAME.toLowerCase()]
+  }
+  // first address of xff, list is comma separated
   if (ipAddress && ipAddress.indexOf(',') > -1) {
     ipAddress = ipAddress.split(',')[0]
+  } else if (ipAddress === '') {
+    ipAddress = req.ip
```

Comment:
Are you sure this gets IP from `X-Forwarded-For`? or it should be such value

```
When the trust proxy setting does not evaluate to false, the value of this property is derived from the left-most entry in the X-Forwarded-For header. This header can be set by the client or by the proxy.
```

Reply:
I think this is the behavior we want, if TRUST_PROXY is truthy but IP_HEADER_PROP_NAME value isn't set ipAddress ends up being null, causing errors. Before, if TRUST_PROXY was not set we would alse end up with a null value. Express defaults to x-forwarded-for, and I have confirmed that x-forwarded-for is the default in UAT.

In the end this should prevent having a null IP being set, falling back to what express thinks is the IP. But it does raise a good point that there should be a warning in the logs if it couldn't get the IP.

Reply:
ok, just to confirm